core/src/main/java/com/linecorp/armeria/client/routing/ZookeeperEndpointGroup.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.function.Function;
+
+import org.apache.zookeeper.AsyncCallback.StatCallback;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.KeeperException.Code;
+import org.apache.zookeeper.WatchedEvent;
+import org.apache.zookeeper.Watcher;
+import org.apache.zookeeper.Watcher.Event.KeeperState;
+import org.apache.zookeeper.ZooKeeper;
+import org.apache.zookeeper.data.Stat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.Endpoint;
+
+/**
+ *  a zookeeper {@link EndpointGroup} implementation .
+ *  it will get the {@link EndpointGroup} information from a zookeeper zNode, any update to it will
+ *  reflect asynchronously to the system
+ */
+public class ZookeeperEndpointGroup implements EndpointGroup, Watcher, StatCallback {
+
+    private static final Logger logger = LoggerFactory.getLogger(ZookeeperEndpointGroup.class);
+
+    private String hostPort;
+    private String znode;
+    private List<Endpoint> endpointList;
+    private ZooKeeper zooKeeper;
+    byte[] prevData;
+    Function<byte[], List<Endpoint>> nodeValueToEndpointList;
+
+    final CountDownLatch connectionLatch = new CountDownLatch(1);
+
+    /**
+     * create a zookeeper endpoint group
```

Comment:
- create -> Creates (note the capital C)
- zookeeper -> ZooKeeper
- endpoint group -> `{@link EndpointGroup}.` (note the trailing period)
- Insert an empty line before the first `@param` tag.